astropy/units/quantity.py

```diff
@@ -1166,18 +1166,19 @@ def to_string(self, unit=None, precision=None, format=None, subfmt=None):
             },
         }
 
+        # Set the precision, if set, otherwise use numpy default
+        pops = np.get_printoptions()
```

Comment:
Yes, it makes sense that this doesn't work with arrays. I fear we'll need to explicitly set one of the numpy formatting arguments (maybe just precision itself? but a different default?), since those will be used if one just expands `{self.value}`. I remember the machinery changed a little while ago, so probably good to start by looking what it presently allows.

Reply:
Should be fixed - while being backwards compatible - by using `np.array2string` in the conditional

```
        if format not in formats:
            raise ValueError(f"Unknown format '{format}'")
        elif format is None:
            if precision is None:
                # Use default formatting settings
                return f'{self.value}{self._unitstr:s}'
            else:
                # np.array2string properly formats arrays as well as scalars
                return f'{np.array2string(self.value, precision=precision, floatmode="fixed")}{self._unitstr:s}'
```

The `CI/Python3.8` job is failing for an unknown reason. Locally, it works perfectly fine. In any case, I will mark this PR as ready.